src/uucore/src/lib/parser/parse_date_modifier.rs

```diff
@@ -0,0 +1,328 @@
+                ExpectNum => match self.parse_num() {
+                    Ok(num) => {
+                        curr_num = num;
+                        self.state = ExpectChrono;
+                    }
+                    Err(EndOfString) => {
+                        return Ok(map);
+                    }
+                    Err(e) => {
+                        return Err(e);
+                    }
+                },
+                ExpectChrono => {
+                    match self.parse_unit() {
+                        Ok(chrono) => {
+                            if map.contains_key(&chrono) {
+                                *map.get_mut(&chrono).unwrap() += curr_num;
+                            } else {
+                                map.insert(chrono, curr_num);
+                            }
+                            self.state = ExpectNum;
+                        }
+                        Err(err) => {
+                            return Err(err);
+                        }
+                    }
+                }
+            }
+        }
+        Ok(map)
+    }
+
+    fn parse_num(&mut self) -> Result<i64, ParseError> {
+        while self.cursor < self.haystack.len() && self.haystack[self.cursor].is_ascii_whitespace()
+        {
+            self.cursor += 1;
+        }
+        if self.cursor >= self.haystack.len() {
+            return Err(EndOfString);
+        }
+        let bytes = &self.haystack[self.cursor..];
+        if bytes[0] == b'+' || bytes[0] == b'-' || (bytes[0] > 47 && bytes[0] <= 57) {
+            let mut nums = vec![bytes[0] as char];
+            let mut i = 1;
+            loop {
+                if i >= bytes.len() {
+                    break;
+                }
+                if let n @ 48..=57 = bytes[i] {
+                    nums.push(n as char);
+                    if bytes[i].is_ascii_whitespace() {
+                        self.cursor += 1;
+                        break;
+                    }
+                    self.cursor += 1;
+                    i += 1;
+                } else if bytes[i].is_ascii_whitespace() {
+                    self.cursor += 1;
+                    i += 1;
+                    continue;
+                } else {
+                    self.cursor += 1;
+                    break;
+                }
+            }
+            let n_as_string = nums.iter().collect::<String>();
+            n_as_string.parse::<i64>().map_err(|_| IntOverflow)
+        } else {
+            Err(UnexpectedToken)
+        }
+    }
+
+    fn parse_unit(&mut self) -> Result<ChronoUnit, ParseError> {
+        while self.cursor < self.haystack.len() && self.haystack[self.cursor].is_ascii_whitespace()
+        {
+            self.cursor += 1;
+        }
+        if self.cursor >= self.haystack.len() {
+            return Err(NoUnitPresent);
+        }
+        let bytes = &self.haystack[self.cursor..].to_ascii_lowercase();
+        match bytes[0] {
+            b'd' => {
```

Comment:
i can probably by simplified by something like:

```
    let units = [
        (b'd', "day", ChronoUnit::Day),
        (b'w', "week", ChronoUnit::Week),
        (b'm', "month", ChronoUnit::Month),
        (b'y', "year", ChronoUnit::Year),
        (b'h', "hour", ChronoUnit::Hour),
        (b's', "second", ChronoUnit::Second),

    ];
    for &(byte, unit_str, chrono_unit) in &units {
        if bytes.starts_with(unit_str) {
            self.cursor += unit_str.len();
            return Ok(chrono_unit);
        }
    }
```

Reply:
Thank you for reviewing this PR I can implement the changes later this evening.